src/middlewared/middlewared/plugins/etc.py

```diff
@@ -366,11 +346,20 @@ def opener(path, flags):
         if outfile_dirname != '/etc':
             os.makedirs(outfile_dirname, exist_ok=True)
 
-        with open(full_path, "w", opener=opener) as f:
-            perms_changed = self.set_etc_file_perms(f.fileno(), entry)
-            contents_changed = write_if_changed(f.fileno(), rendered)
+        payload = self.get_perms_and_ownership(entry)
+        try:
+            changes = write_if_changed(full_path, rendered, raise_error=True, **payload)
+        except UnexpectedFileChange as e:
```

Comment:
Should we just log this in the `write_if_changed` instead of handling this as an exception? I feel like this try/except will be reused a lot. Also raising an exception has a connotation that the changes were not written.

Reply:
That's reasonable.